File: 05-Tipagens/05-Tipagens/05-Exerci╠ücios/01-Co╠üdigos incorretos/08-Klotin/exercise.kt
class Person(val name: String, val age: Int)

fun printPersonDetails(person: Person) {
    println("Name: ${person.name}, Age: ${person.age}")
}

fun main() {
    val person = Person("John", 25)
    printPersonDetails(person)

    val incorrectAge: String = "Twenty-five"
    printPersonDetails(incorrectAge)

    val number: Int = "Ten"
    println(number)
}

File: tipagens/05-Tipagens/05-Exerci╠ücios/01-Co╠üdigos incorretos/06-TypeScript/exercise.ts
class Person {
    constructor(public name: string, public age: number) {}
}

function greet(person: Person) {
    console.log(`Hello, ${person.name}. You are ${person.age} years old.`);
}

const person = new Person("Alice", 30);
greet(person);

const notPerson: string = "This is a string";
// greet(notPerson); greet só recebe classe

// const number: number = "This is a string";
